sound/soc/sof/ipc4-topology.c

```diff
@@ -498,6 +530,16 @@ static int sof_ipc4_widget_setup_msg(struct snd_sof_widget *swidget, struct sof_
 	msg->extension = SOF_IPC4_MOD_EXT_CORE_ID(swidget->core);
 
 	type = (fw_module->man4_module_entry.type & SOF_IPC4_MODULE_DP) ? 1 : 0;
+	switch (swidget->comp_domain) {
+	case SOF_COMP_DOMAIN_LL:
+		type = 0;
+		break;
+	case SOF_COMP_DOMAIN_DP:
+		type = 1;
+		break;
+	default:
+		break;
+	}
```

Comment:
should this be:

```c
	switch (swidget->comp_domain) {
	case SOF_COMP_DOMAIN_LL:
		type = 0;
		break;
	case SOF_COMP_DOMAIN_DP:
		type = 1;
		break;
	default:
		type = (fw_module->man4_module_entry.type & SOF_IPC4_MODULE_DP) ? 1 : 0;
		break;
	}
```
?

Reply:
I don't see where it was asked, for me it looks more intuitive to query -the type from manifest only if it is not going to be changed anyways from tplg.

any module can be moved in/out of a domain freely?